qiskit/ignis/experiments/calibration/cal_base_analysis.py

```diff
@@ -0,0 +1,348 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class of calibration analysis."""
+
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, Optional, Any, Iterator, List, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from qiskit.result import Result, Counts
+from scipy import optimize
+
+from qiskit.ignis.experiments.base import Analysis
+from qiskit.ignis.experiments.calibration.cal_metadata import CalibrationMetadata
+from qiskit.ignis.experiments.calibration.data_processing import DataProcessingSteps
+
+
+class BaseCalibrationAnalysis(Analysis):
+    """
+    A base analysis for calibration.
+    """
```

Comment:
We can implement multi-dimensional by overwriting some of methods (because the data structure is dimension-less data frame, we can construct arbitrary vector for fitting). Perhaps it's better to define 1d fitter though.